src/test/java/eu/europeana/metis/sandbox/service/metrics/MetricsServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+package eu.europeana.metis.sandbox.service.metrics;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.common.Step;
+import eu.europeana.metis.sandbox.entity.DatasetStatistic;
+import eu.europeana.metis.sandbox.entity.StepStatistic;
+import eu.europeana.metis.sandbox.entity.problempatterns.DatasetProblemPatternStatistic;
+import eu.europeana.metis.sandbox.repository.RecordLogRepository;
+import eu.europeana.metis.sandbox.repository.RecordRepository;
+import eu.europeana.metis.sandbox.repository.problempatterns.DatasetProblemPatternRepository;
+import eu.europeana.patternanalysis.view.ProblemPatternDescription.ProblemPatternId;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Unit tests for {@link MetricsServiceImpl} class
+ */
+@ExtendWith(MockitoExtension.class)
+class MetricsServiceImplTest {
+
+  @Mock
+  private RecordRepository recordRepository;
+  @Mock
+  private RecordLogRepository recordLogRepository;
+  @Mock
+  private DatasetProblemPatternRepository problemPatternRepository;
+  @Spy
+  private MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+  @InjectMocks
+  private MetricsServiceImpl metricsService;
+
+  private void assertRepositoriesAndMeterRegistry() {
+    verify(recordRepository, times(2)).getMetricDatasetStatistics();
+    verify(recordLogRepository, times(2)).getMetricStepStatistics();
+    verify(problemPatternRepository, times(2)).getMetricProblemPatternStatistics();
+    assertEquals(40, meterRegistry.getMeters().size());
+  }
+
+  @Test
+  void processMetrics() {
+    when(recordRepository.getMetricDatasetStatistics()).thenReturn(List.of(new DatasetStatistic("1", 10L)));
+    when(recordLogRepository.getMetricStepStatistics()).thenReturn(
+        List.of(new StepStatistic(Step.HARVEST_ZIP, Status.SUCCESS, 10L)));
+    when(problemPatternRepository.getMetricProblemPatternStatistics()).thenReturn(List.of(
+        new DatasetProblemPatternStatistic(ProblemPatternId.P2.name(), 5L)));
+
+    metricsService.processMetrics();
+
+    assertRepositoriesAndMeterRegistry();
+  }
+
+  @Test
+  void processMetrics_whenDatasetNull() {
+    when(recordRepository.getMetricDatasetStatistics()).thenReturn(null);
+    metricsService.processMetrics();
+
+    assertEquals(0L, metricsService.getTotalRecords());
```

Comment:
Shouldn't we be testing the results that are contained in the metric registry instead?
I think those methods should be private and we should only check the aggregation of the values that are in the metric registry.